File: ts/easy/numIdenticalPairs.ts
/*
  Given an array of integers nums, return the number of good pairs.
  A pair (i, j) is called good if nums[i] == nums[j] and i < j.
*/

function numIdenticalPairs(nums: number[]): number {
  const result: number[][] = [];
  for (let i = 0; i < nums.length; i++) {
      for (let j = 1; j < nums.length; j++) {
          if (i < j && nums[i] === nums[j]) {
              result.push([nums[i], nums[j]]);
          }
      }
  }
  return result.length;
};

File: python/easy/num_identical_pairs.py
from typing import List

"""
  Given an array of integers nums, return the number of good pairs.
  A pair (i, j) is called good if nums[i] == nums[j] and i < j.
"""

def num_identical_pairs(nums: List[int]) -> int:
    counter = 0
    hash_table = {}

    for num in nums:
        counter += hash_table.get(num, 0)
        hash_table[num] = hash_table.get(num, 0) + 1
    return counter
